containers/event-db-migrations/entry.sh

```diff
@@ -1,46 +1,147 @@
+# ADMIN_FIRST_NAME - The first name of the admin user for graphql (optional)
+# ADMIN_LAST_NAME - The last name of the admin user for graphql (optional)
+# ADMIN_ABOUT - The about of the admin user for graphql (optional)
+# ADMIN_EMAIL - The email of the admin user for graphql (optional)
+# REINIT_EVENT_DB - If set, the database will be reinitialized (optional) (DESTRUCTIVE)
+# SKIP_EVENT_DB_INIT - If set, the event database will not be initialized (optional)
+# SKIP_GRAPHQL_INIT - If set, graphql will not be initialized (optional)
+# DEBUG - If set, the script will print debug information (optional)
+# DEBUG_SLEEP - If set, the script will sleep for the specified number of seconds (optional)
+# ---------------------------------------------------------------
+
+set +x
+set -o errexit
+set -o pipefail
+set -o nounset
+set -o functrace
+set -o errtrace
+set -o monitor
+set -o posix
+shopt -s dotglob
+
+check_env_vars() {
+    local env_vars=("$@")
+
+    # Iterate over the array and check if each variable is set
+    for var in "${env_vars[@]}"; do
+        if [ -z "${!var}" ]; then
+            echo ">>> Error: $var is required and not set."
+            exit 1
+        fi
+    done
+}
+
+debug_sleep() {
+    if [ -n "${DEBUG_SLEEP:-}" ]; then
+        echo "DEBUG_SLEEP is set. Sleeping for ${DEBUG_SLEEP} seconds..."
+        sleep "${DEBUG_SLEEP}"
+    fi
+}
+
+echo ">>> Starting entrypoint script..."
+
+# Check if all required environment variables are set
+REQUIRED_ENV=(
+    "DB_HOST"
+    "DB_PORT"
+    "DB_NAME"
+    "DB_SUPERUSER"
+    "DB_SUPERUSER_PASSWORD"
+    "DB_USER"
+    "DB_USER_PASSWORD"
+    "ADMIN_ROLE_PASSWORD"
+    "ADMIN_USER_PASSWORD"
+    "ANON_ROLE_PASSWORD"
+)
+check_env_vars "${REQUIRED_ENV[@]}"
 
-# Check if the database has been initialized
-if [ ! -f ./tmp/initialized.txt ] || [ "$reinitDb" = true ]; then
-  echo "Initializing database..."
-  psql -e -U $PGUSER -d $PGDATABASE -h $PGHOST -p $PGPORT -f ./setup/setup-db.sql \
-    -v dbName=CatalystEventDev \
-    -v dbDescription="Local Dev Catalayst Event DB" \
-    -v dbUser="catalyst-event-dev"
-  echo "Initializing database for graphql..."
-  psql -e -U $PGUSER -d $PGDATABASE -h $PGHOST -p $PGPORT -f ./setup/graphql-setup.sql \
-    -v dbName=CatalystEventDev \
-    -v dbUser="catalyst-event-dev" \
-    -v adminUserFirstName="Admin" \
-    -v adminUserLastName="Default" \
-    -v adminUserAbout="Default Admin User" \
-    -v adminUserEmail="admin.default@projectcatalyst.io"
-
-  touch ./tmp/initialized.txt
+# Export environment variables
+export PGHOST="${DB_HOST}"
+export PGPORT="${DB_PORT}"
+export PGUSER="${DB_SUPERUSER}"
+export PGPASSWORD="${DB_SUPERUSER_PASSWORD}"
+export PGDATABASE="${DB_NAME}"
+
+: "${ADMIN_FIRST_NAME:='Admin'}"
+: "${ADMIN_LAST_NAME:='Default'}"
+: "${ADMIN_ABOUT:='Default Admin User'}"
+: "${ADMIN_EMAIL:='admin.default@projectcatalyst.io'}"
+
+# Sleep if DEBUG_SLEEP is set
+debug_sleep
+
+# Initialize database if necessary
```

Comment:
I don't know why we need `/tmp/initialize` at all?

We should be able to run the graphql setup multiple times without concern.
The ONLY issue with the graphql setup is deleting the user table.

I propose we just have the env vars which control initing the base DB and graphql ONLY.  
I also think it shouldn't be a SKIP env var it should be a "INIT_EVENT_DB" and "INIT_GRAPHQL" env var,  default to false.

Reply:
The container will be configured as a Kubernetes Job that will be deployed with the rest of the applications. Since it's a job, it will only run once. However, there is a small chance of it inadvertently being run again (due to operator error), and if it's been configured with `INIT_EVENT_DB`, then the event database gets wiped. Whereas, assuming the volume is mounted properly, with the `/tmp/initialize` approach the script would refuse to initialize the database unless the `REINIT_EVENT_DB` environment variable was given. 

So, by default we initialize, unless we already know it's initialized. I'm not a fan of using a volume for tracking state. If you know of a better way to determine when we should initialize, I would prefer a better approach.

Reply:
Yeah, that's sort of what I was trying to get at in Slack. We would need something to validate that both initializations ran. For example, maybe for the `event_db` initialization we can verify the expected user/database exists. For the graphql stuff, maybe just look for the `private` schema?

@stevenj Thoughts?

Reply:
Yes, something exactly like this.

Reply:
For the DB the very last thing we could do when initialising it is write a record to the `config` table eg:
```
`id`: "Initialized", id2: "", id3: "", value: {"at" : "<datetime>"}
```

Because this is the VERY last thing we do when we init the database, we know its been done if that record exists.
If it stops at any point before that the DB is screwed and needs to be re-initialized anyway.

We don;t need to check the value of the `value` just that we can select this record in the `config` table and it returns data.

Reply:
Because graphql init is needed,  we don't really need a separate check for it.  IF we have this record graphql was setup along with everything else.

Reply:
We can tackle this in another PR.